nebula-logger/core/main/logger-engine/lwc/logger/logEntryBuilder.js

```diff
@@ -107,28 +95,40 @@ const LogEntryBuilder = class {
     }
 
     _logToConsole() {
-        if (!this.isConsoleLoggingEnabled) {
-            return;
-        }
+        this.#settingsPromise().then(setting => {
+            this.isConsoleLoggingEnabled = setting.isConsoleLoggingEnabled;
 
-        /* eslint-disable no-console */
-        switch (this.loggingLevel) {
-            case 'ERROR':
-                console.error(this.message, this);
-                break;
-            case 'WARN':
-                console.warn(this.message, this);
-                break;
-            case 'INFO':
-                console.info(this.message, this);
-                break;
-            default:
-                console.debug(this.message, this);
-                break;
-        }
+            if (!this.isConsoleLoggingEnabled) {
+                return;
+            }
+
+            const consoleMessagePrefix = '%c Nebula Logger ';
+            const consoleFormatting = 'background: #0c598d; color: #fff; font-size: 12px; font-weight:bold;';
+            let consoleLoggingFunction;
+            switch (this.loggingLevel) {
+                case 'ERROR':
+                    consoleLoggingFunction = console.error;
+                    break;
+                case 'WARN':
+                    consoleLoggingFunction = console.warn;
+                    break;
+                case 'INFO':
+                    consoleLoggingFunction = console.info;
+                    break;
+                default:
+                    consoleLoggingFunction = console.debug;
+                    break;
+            }
+
+            consoleLoggingFunction(consoleMessagePrefix, consoleFormatting, this.message);
+            console.groupCollapsed(consoleMessagePrefix, consoleFormatting, 'Details for: ' + this.message);
+            consoleLoggingFunction(JSON.parse(JSON.stringify(this)));
```

Comment:
The cloning `JSON.parse(JSON.stringify`, is never perfect (it does not copy functions, and fails on cyclic objects), so it might be worth a comment to clarify the intent (especially because it is about the output into the browser.

Reply:
Great suggestion, I'll add a comment to document this - long-term, there may be better way(s) to print out the data, but a comment for now should help.